src/actions/searchFromIsbn.test.ts

```
@@ -0,0 +1,186 @@
+    const mockAuthApi = getAuth(mockDb).api;
+    (mockAuthApi.getSession as ReturnType<typeof vi.fn>).mockResolvedValue(null);
+
+    const result = await searchFromIsbn('9781234567890');
+    expect(result).toBeUndefined();
+    expect(searchBooksFromNDL).not.toHaveBeenCalled();
+  });
+
+  it('should return undefined if NDL search finds no books', async () => {
+    // Mock getSession to return a user
+    const mockAuthApi = getAuth(mockDb).api;
+    (mockAuthApi.getSession as ReturnType<typeof vi.fn>).mockResolvedValue({ user: mockUser, session: {} as any });
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)